### PR TITLE
feature(shell): new echo command to display fmt strings

### PR DESCRIPTION
**Pull request by rgs1**
Signed-off-by: Raul Gutierrez S <rgs@itevenworks.net>

### CODE REVIEW

File: zk_shell/tests/test_basic_cmds.py [hunk 1]
```diff
@@ -386,3 +386,8 @@ def test_session_info(self):
         lines = [line for line in self.output.getvalue().split("\n") if line != ""]
         self.assertEqual(1, len(lines))
         self.assertIn("sessionid", self.output.getvalue())
+
+    def test_echo(self):
+        self.shell.onecmd("create %s/jimeh gimeh" % (self.tests_path))
```
Comment (Yasumoto) [inline zk_shell/tests/test_basic_cmds.py]: Optional, but typically it's `jimmeh` and `gimmeh`.

File: zk_shell/shell.py [hunk 2]
```diff
@@ -2778,6 +2778,35 @@ def complete_reconfig(self, cmd_param_text, full_cmd, *rest):
         completers = [complete_cmd, complete_arg, complete_config]
         return complete(completers, cmd_param_text, full_cmd, *rest)
 
+    @ensure_params(Required("fmtstr"), MultiOptional("cmds"))
+    def do_echo(self, params):
+        """
+\x1b[1mNAME\x1b[0m
```
Comment (Yasumoto) [inline zk_shell/shell.py]: Optional, but have you considered a library that gives you these color codes ~for free?

**Comment by Yasumoto** [comment]
Two questions/suggestions, otherwise \#ship!

File: zk_shell/tests/test_basic_cmds.py (hunk 1 above)
Comment (rgs1) [inline zk_shell/tests/test_basic_cmds.py]: Good point.

File: zk_shell/shell.py (hunk 2 above)
Comment (rgs1) [inline zk_shell/shell.py]: Well, the thing is that the cmd module requires these to be docstrings. How can we generate docstrings dynamically?

Comment (Yasumoto) [inline zk_shell/shell.py]: Here's what I was thinking, but up to you with whatever looks better.

Something like https://github.com/tartley/colorama#colored-output:

```
"""
%sNAME%s
A cool name.
""" % (Fore.RED, Style.RESET_ALL)

Comment (rgs1) [inline zk_shell/shell.py]: I don't think extrapolation works for docstrings:

```
>> def bar():
    """
    i am bar
    """
    pass
>> print bar.__doc__
'\n    i am bar\n    '
>>def foo():
    """
    i am %s
    """ % ('foo')
    pass
>> print foo.__doc__
(empty)
```
Is this not expected or am I doing it wrong?

**Pull request merged**